source/grid/grid_in.cc

```diff
@@ -704,14 +704,9 @@ void GridIn<dim, spacedim>::read_ucd (std::istream                            &i
           for (unsigned int i=0; i<GeometryInfo<dim>::vertices_per_cell; ++i)
             in >> cells.back().vertices[i];
 
-          // to make sure that the cast won't fail
-          Assert(material_id<= std::numeric_limits<types::material_id>::max(),
-                 ExcIndexRange(material_id,0,std::numeric_limits<types::material_id>::max()));
           // we use only material_ids in the range from 0 to numbers::invalid_material_id-1
-          Assert(material_id < numbers::invalid_material_id,
-                 ExcIndexRange(material_id,0,numbers::invalid_material_id));
-
-          cells.back().material_id = static_cast<types::material_id>(material_id);
+          AssertIndexRange(new_material_id, numbers::invalid_material_id);
```

Comment:
This is now a no-op with the type you use.

Reply:
No, we don't allow `new_material_id` to be `unsigned int(-1)`. All the other checks later are no-ops and coverity is complaining about this.

Reply:
Then, we should really read in `int`s here, shouldn't we?

Reply:
Yes -- isn't that what the code was doing before? ;-)

Reply:
```
// we use an unsigned int because we
// fill this variable through an read-in process
unsigned int material_id;

in >> dummy          // cell number
   >> material_id;
```
No, we are using `unsigned int`s and the conversion is done implicitly as far as I see.

Reply:
Ah, I see what you mean. I have no idea whether it's specified as signed or unsigned int. Either is fine by me.

Reply:
Let me address this in another PR.